tgui/packages/tgui/interfaces/PowerMonitor/type.ts

```diff
@@ -67,3 +68,45 @@ export interface PowerMonitorSmesStaticItemData {
 export const isDataForApc = (data: PowerMonitorData): data is PowerMonitorApcData => data.type === PowerMonitorType.Apc;
 export const isDataForSmes = (data: PowerMonitorData): data is PowerMonitorSmesData =>
   data.type === PowerMonitorType.Smes;
+
+export enum ApcTableHeaderColumns {
+  Area = 0,
+  Equipment = 1,
+  Lighting = 2,
+  Environment = 3,
+  Load = 4,
+  CellCharge = 5,
+  CellState = 6
+}
+
+export type ApcTableHeaderColumnSortState = {
+  dir: SortDirection,
+  field: ApcTableHeaderColumns
+}
```

Comment:
```suggestion
export type ApcTableSortState = SingleFieldSortState<ApcTableHeaderColumns>;
```
Then add to common/sorting/type.ts:
```
export interface SingleSortState<T> {
  dir: SortDirection;
  field: T;
}
```

Reply:
Implemented with the rest of the improvements, I had some fun learning generics in typescript.